Mathlib/Topology/UniformSpace/CompactConvergence.lean

```diff
@@ -223,6 +221,29 @@ theorem mem_compactConvergence_entourage_iff (X : Set (C(α, β) × C(α, β)))
   simp [hasBasis_compactConvergenceUniformity.mem_iff, and_assoc]
 #align continuous_map.mem_compact_convergence_entourage_iff ContinuousMap.mem_compactConvergence_entourage_iff
 
+theorem _root_.CompactExhaustion.hasBasis_compactConvergenceUniformity {ι : Type*}
+    {p : ι → Prop} {V : ι → Set (β × β)} (K : CompactExhaustion α) (hb : (𝓤 β).HasBasis p V) :
+    HasBasis (𝓤 C(α, β)) (fun i : ℕ × ι ↦ p i.2) fun i ↦
+      {fg | ∀ x ∈ K i.1, (fg.1 x, fg.2 x) ∈ V i.2} := by
```

Comment:
Unless I'm missing something, I think this is already true at the level of `UniformOnFun` without more work. That said, we may want a more general result in that setup, so do as you want.

Reply:
What result is true for `UniformOnFun`?

Reply:
For `α →ᵤ[{K | IsCompact K}] β` all of your results are already true (both the explicit basis and the countably generated uniformity). More generally, if `𝔖` is an ideal and you have a basis of your ideal, then you have the same basis result for `α →ᵤ[𝔖] β` (eventually [UniformOnFun.hasBasis_uniformity_of_basis](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/UniformConvergenceTopology.html#UniformOnFun.hasBasis_uniformity_of_basis) should be stated this way), hence if `𝔖` is a countably generated ideal and `𝓤 β` is a countably generated filter then `α →ᵤ[𝔖] β` has countably generated uniformity.

Reply:
Added relevant generalizations.